### ISSUE TITLE
Window viewport height resizing back/forth one pixel below title bar on low_processor_mode update

### ISSUE THREAD

**Issue by lostminds** [godotengine/godot#85742]
### Godot version

v4.2.stable.official [46dc27791]

### System information

Apple M2 Pro, macOS 14.1.1

### Issue description

I've encountered a subtle graphical issue, where it seems the godot viewport and macOS window is fighting resizing the window content view one pixel back/forth below the window title bar. As if some system is incorrectly getting the height of the title bar and resizing the content view, then some other system is resizing it back based on a different title bar height.

The reason I've seen it is that I'm building an editor app where I've enabled the `application/run/low_processor_mode` setting, so that the window only updates when something actually changes. This makes the "height fighting" described above visible since apparently one of the resizes is caused by something that still triggers so you can see one such resize and then see godot resize the viewport back next time it updates (I think). For example you can trigger this by selecting and deselecting the window, hovering over the title bar or content.

The issue is not visible in projects that are updating continuously (as in not low_processor_mode), but on careful inspection you can see that it occurs once at startup with the same one pixel resize under the titlebar. So my fear is that it could be happening in these projects as well, but that the resizes are happening back and forth before they are displayed on screen so they're not visible to the user.

A curious thing however is that this seems to be very low level, making it hard to inspect or document, because even if it's apparent on screen when looking at it the issue goes away if I try to do a screen recording or even grabbing a screenshot. I'm guessing because the recording is somehow triggering the area to refresh in the system window server before the screen content is grabbed. So you'll have to make due with this very shaky mobile recording to show the issue:

https://github.com/godotengine/godot/assets/17763524/1d4d81d6-510f-441e-9086-e9bf123b4ba1

Trying to look at the main viewport.size property it's not changing either, so if the context is resizing it's happening lower down and not affecting the high level godot viewport object. 

In my testing I'm getting the issue in a new project on any Vulkan backed renderer (as in Mobile, Forward + or the ANGLE Compatibility renderer). However, it does not occur when using the old OpenGL Compatibility renderer.

It might be that this is just a pretty harmless visual glitch, or possibly a macOS bug. But if the viewport or underlying vulkan context is being resized back and forth, possibly quite often, this could potentially be causing more serious performance issues reported like https://github.com/godotengine/godot/issues/85619 for 4.2 

### Steps to reproduce

- Create a new project (any renderer) on macOS
- Set application/run/low_processor_mode
- Set up a default scene so it can run (just an empty control node for example)
- Run and observe viewport resizing one pixel in height (white) under window title bar back and forth as you select/deselect/hover the application window
- Optionally, change to Compatibility renderer Mode with the old OpenGL driver (not ANGLE) run again and observe the issue doesn't happen. 

### Minimal reproduction project

A new blank project reproduces the issue for me, just requires setting the low_processor_mode setting to see

**Comment by bruvzg** [comment]
I can't reproduce it. What display (and resolution / scaling) are you using? I suspect it might be related to hiDPI scaling.

**Comment by lostminds** [comment]
Yes, I'm using two high DPI displays, set to 1920x1080 x2 retina (so 4k). Disabling highDPI support in the Godot project does not fix the issue. However, changing one monitor to the native 3840x2160 resolution (so same monitor, but no resolution-scaling) the issue doesn't seem to appear on that monitor. But if I move it over again to the other monitor it appears there again. 

So it seems you're probably right that it's somehow related to the highDPI monitor resolution scaling. 

Another thing I noticed is that when you look closely at screenshots it seems that macOS now renders a small subtle shadow gradient on top of the content view below the title bar in retina mode. This looks like corresponds to the 1-2 px that goes white. I also noticed that setting the window to borderless I don't see the issue, but I guess that just says it has something to do with the window titlebar.

![Screenshot 2023-12-05 at 14 27 05](https://github.com/godotengine/godot/assets/17763524/2f82ca16-9588-4ea9-8dbe-e4df7854dcc0)

So perhaps it could be just a bug with the update where the window server is trying to composit that shadow on the content, but for some reason it's not getting the correct content from the window to draw it on? The same shadow does however exist in regular resolutions as well, only it's just one pixel there and that seems to work fine.
